src/pybind/mgr/dashboard/rest_client.py

```diff
@@ -0,0 +1,509 @@
+                    "{}"  # TODO remove
+                    .format(self.client_name, resp.status_code, pf(
+                        resp.content)),
+                    resp.status_code,
+                    resp.content)
+        except ConnectionError as ex:
+            if ex.args:
+                if isinstance(ex.args[0], SSLError):
+                    errno = "n/a"
+                    strerror = "SSL error. Probably trying to access a non " \
+                               "SSL connection."
+                    logger.error("%s REST API failed %s, SSL error.",
+                                 self.client_name, method.upper())
+                else:
+                    match = re.match(r'.*: \[Errno (-?\d+)\] (.+)',
+                                     ex.args[0].reason.args[0])
+                    if match:
+                        errno = match.group(1)
+                        strerror = match.group(2)
+                        logger.error(
+                            "%s REST API failed %s, connection error: "
+                            "[errno: %s] %s",
+                            self.client_name, method.upper(), errno, strerror)
+                    else:
+                        errno = "n/a"
+                        strerror = "n/a"
+                        logger.error(
+                            "%s REST API failed %s, connection error.",
+                            self.client_name, method.upper())
+            else:
+                errno = "n/a"
+                strerror = "n/a"
+                logger.error("%s REST API failed %s, connection error.",
+                             self.client_name, method.upper())
+
+            if errno != "n/a":
+                ex_msg = (
+                    "{} REST API cannot be reached: {} [errno {}]. "
+                    "Please check your configuration and that the API endpoint"
+                    " is accessible"
+                    .format(self.client_name, strerror, errno))
+            else:
+                ex_msg = (
+                    "{} REST API cannot be reached. Please check "
+                    "your configuration and that the API endpoint is"
+                    " accessible"
+                    .format(self.client_name))
+            raise RequestException(
+                ex_msg, conn_errno=errno, conn_strerror=strerror)
+        except InvalidURL as ex:
+            logger.exception("%s REST API failed %s: %s", self.client_name,
+                             method.upper(), str(ex))
+            raise RequestException(str(ex))
+
+    @staticmethod
+    def api(path, **api_kwargs):
+        def call_decorator(func):
+            def func_wrapper(self, *args, **kwargs):
+                method = api_kwargs.get('method', None)
+                resp_structure = api_kwargs.get('resp_structure', None)
+                args_name = inspect.getargspec(func).args
+                args_dict = dict(itertools.izip(args_name[1:], args))
```

Comment:
`itertools.izip` does not exist in python 3. Please use just the built-in function `zip`, which will work in both python 2 and 3, or add a conditional based on the python version to either  use `itertools.izip` in python 2 and `zip` in python 3.